lib/task/tools/expireDataTask.class.php

```diff
@@ -0,0 +1,165 @@
+  }
+
+  protected function execute($arguments = array(), $options = array())
+  {
+    parent::execute($arguments, $options);
+
+    // Abort if data type isn't supported
+    if (!in_array(strtolower($arguments['data-type']), array_keys(self::$TYPE_SPECIFICATONS)))
+    {
+      throw new sfException('Aborted: unsupported data type.');
+    }
+
+    $typeSpec = self::$TYPE_SPECIFICATONS[($arguments['data-type'])];
+
+    // Set older than option if not set and a non-zero maximum age is set for data type
+    $checkAgeSetting = !isset($options['older-than']) && isset($typeSpec['age_setting_name']);
+    if ($checkAgeSetting && $maxAge = sfConfig::get('app_clipboard_save_max_age'))
+    {
+      // Throw error if setting value isn't an integer
+      if (!ctype_digit($maxAge))
+      {
+        throw new sfException(
+          sprintf('Error: setting %s value "%s" is non-numeric.', $typeSpec['age_setting_name'], $maxAge)
+        );
+      }
+
+      // Use date type's maximum age setting to calculate older than option
+      $date = new DateTime();
+      $interval = new DateInterval('P'. sfConfig::get('app_clipboard_save_max_age') .'D');
+      $date->sub($interval);
+      $options['older-than'] = $date->format('Y-m-d');
+
+      // Let user know that date type's maximum age setting was used
+      $this->logSection(
+        'expire-data',
+        sprintf('Used %s setting to set expiry date of %s.', $typeSpec['age_setting_name'], $options['older-than'])
+      );
+    }
+
+    // Abort if not forced or confirmed
+    if (!$options['force'] && !$this->getConfirmation($options, $typeSpec['plural_name']))
+    {
+      $this->logSection('expire-data', 'Aborted.');
+      return;
+    }
+
+    // Expire data and report results
+    $methodName = $arguments['data-type'] .'ExpireData';
+    $deletedCount = $this->$methodName($options);
+    $this->logSection('expire-data', sprintf('Finished! %d saved clipboards deleted.', $deletedCount));
+  }
+
+  private function getConfirmation($options, $typeNamePlural)
+  {
+    $message = 'Are you sure you want to delete ';
+
+    if (isset($options['older-than']))
+    {
+      $message .= sprintf('%s older than %s?', $typeNamePlural, $options['older-than']);
+    }
+    else
+    {
+      $message .= sprintf('all %s?', $typeNamePlural);
+    }
+
+    return $this->askConfirmation($message, 'QUESTION_LARGE', false);
+  }
+
+  private function clipboardExpireData($options)
+  {
+    // Assemble criteria
+    $criteria = new Criteria;
+
+    if (isset($options['older-than']))
+    {
+      $criteria->add(QubitClipboardSave::CREATED_AT, $options['older-than'], Criteria::LESS_THAN);
+    }
+
+    // Delete clipbooard saves and save items
+    $deletedCount = 0;
+
+    foreach(QubitClipboardSave::get($criteria) as $save)
+    {
+      $itemCriteria = new Criteria;
+      $itemCriteria->add(QubitClipboardSaveItem::SAVE_ID, $save->id);
+
+      foreach(QubitClipboardSaveItem::get($itemCriteria) as $item)
```

Comment:
With `onDelete=cascade` this should not be needed

Reply:
Good call.